pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachineUSBDevices/index.vue

```diff
@@ -0,0 +1,229 @@
+      vms:     this.$store.dispatch(`harvester/findAll`, { type: HCI.VM })
+    };
+
+    const res = await allHash(hash);
+
+    for (const key in res) {
+      this[key] = res[key];
+    }
+
+    this.selectedDevices = (this.value?.domain?.devices?.hostDevices || [])
+      .map(({ name }) => name)
+      .filter(name => this.enabledDevices.find(device => device?.metadata?.name === name));
+  },
+
+  data() {
+    return {
+      deviceSchema:  this.$store.getters['harvester/schemaFor'](HCI.USB_DEVICE),
+      deviceHeaders: [
+        { ...STATE },
+        SIMPLE_NAME,
+      ],
+      devices:         [],
+      vms:             [],
+      selectedDevices: [],
+      showMatrix:      false,
+    };
+  },
+
+  watch: {
+    selectedDevices(neu) {
+      const formatted = neu.map((selectedDevice) => {
+        const deviceCRD = this.enabledDevices.find(device => device.metadata.name === selectedDevice);
+        const deviceName = deviceCRD?.status?.resourceName;
+
+        return {
+          deviceName,
+          name: deviceCRD?.metadata.name,
+        };
+      });
+
+      const devices = [
+        ...this.otherDevices(this.value.domain.devices.hostDevices || []),
+        ...formatted,
+      ];
+
+      set(this.value.domain.devices, 'hostDevices', devices);
+    }
+  },
+
+  computed: {
+    deviceOpts() {
+      const filteredOptions = this.enabledDevices.filter((deviceCRD) => {
+        if (this.selectedDevices.length > 0) {
+          const selectedDevice = this.enabledDevices.find(device => device.metadata.name === this.selectedDevices[0]);
+
+          return !this.devicesInUse[deviceCRD?.metadata.name] && deviceCRD.status.nodeName === selectedDevice?.status.nodeName;
+        }
+
+        return !this.devicesInUse[deviceCRD?.metadata.name];
+      });
+
+      return filteredOptions.map((deviceCRD) => {
+        return {
+          value:        deviceCRD?.metadata.name,
+          label:        deviceCRD?.metadata.name,
+          displayLabel: deviceCRD?.status?.description,
+        };
+      });
+    },
+
+    enabledDevices() {
+      return this.devices.filter((device) => {
+        return device.status.enabled;
+      }) || [];
+    },
+
+    devicesInUse() {
+      const inUse = this.vms.reduce((inUse, vm) => {
+        if (vm.metadata.name === this.vm?.metadata?.name) {
+          return inUse;
+        }
+        const devices = get(vm, 'spec.template.spec.domain.devices.hostDevices') || [];
+
+        devices.forEach((device) => {
+          inUse[device.name] = { usedBy: [vm.metadata.name] };
+        });
+
+        return inUse;
+      }, {});
+
+      return inUse;
+    },
+
+    devicesByNode() {
+      const out = {};
+
+      this.enabledDevices.forEach((deviceCRD) => {
+        const nodeName = deviceCRD.status?.nodeName;
+
+        if (!out[nodeName]) {
+          out[nodeName] = [deviceCRD];
+        } else {
+          out[nodeName].push(deviceCRD);
+        }
+      });
```

Comment:
Since you use `?` here, do we need to use `if (!nodeName) { continue }` to skip `undefined`?

Reply:
I think that the null check here is just useless, since the `status` field will be always not null. Anyway I'll write this code better.

Reply:
You are right, It's better to add a null checkof nodeName here.